source/CIPlanetProgressNode.cpp

```diff
@@ -45,33 +40,25 @@ void PlanetProgressNode::draw(const std::shared_ptr<cugl::SpriteBatch>& batch,
 /**
  * Updates the frame of the planet progress animation
  */
-void PlanetProgressNode::update(float timestep) {
+void PlanetProgressNode::update(float timestep, int lockinLayerSize) {
     if (_planetLayer.isLockedIn) {
         setFrame(CANLOCKIN_END);
         return;
     }
     
     if (_planetLayer.layerSize == 0) {
         _currFrame = LOCKIN_END;
-    } else if (_planetLayer.layerSize == 1) {
-        _currFrame = LOCKIN_ONE;
-    } else if (_planetLayer.layerSize == 2) {
-        _currFrame = LOCKIN_TWO;
-    } else if (_planetLayer.layerSize == 3) {
-        _currFrame = LOCKIN_THREE;
-    } else if (_planetLayer.layerSize == 4) {
-        _currFrame = LOCKIN_FOUR;
     } else {
-        _currFrame = LOCKIN_FIVE;
+        _currFrame = round(((float) _planetLayer.layerSize / (float) lockinLayerSize) * PROGRESS_ARC_END);
```

Comment:
I'm not sure if this will work very well because I don't think the circle in the animation progresses linearly

Reply:
I've given it a try with several sizes, and haven't been able to notice much of an issue.